rfc/openfed__setDescription.md

```diff
@@ -0,0 +1,307 @@
+---
+title: "@openfed__setDescription: Propagate specific descriptions to the federated graph"
+author: David Stutt
+---
+
+This RFC proposes a new directive, `@openfed__setDescription`, which aims to deliver the following:
+1. Propagate a specific description to the federated graph.
+2. Propagate a description to the federated graph that is different to the description on the subgraph 
+(internal/private vs. external/public).
+3. Choose not to propagate any descriptions to the federated graph at all.
+
+# Motivations
+
+GraphQL type definitions can (and often do, e.g., entity Objects) exist in multiple subgraphs at any one time—a "shared
+definition".
+Each of these shared definitions allows a description to be defined.
+However, the federated graph can only define a single description.
+In the event that multiple shared definitions define a description, the rules that WunderGraph Cosmo follows are quite 
+simple:
+1. Store the first description encountered.
+2. If a longer description (by character length) is encountered, update the record to this longer description.
+3. If another description of the same length is encountered, keep the current description record (which will be the 
+same length).
+4. Propagate the description in the record to the federated graph.
+
+Customers have requested more granular control over which description should be propagated to the federated graph, 
+if any.
+
+# Proposal
+
+A new directive, `@openfed__setDescription`, aims to facilitate all desired granularity for how and when descriptions
+are propagated to the federated graph.
+
+## Definition
+
+The potential definition of the directive:
+```graphql
+  directive @openfed__setDescription(public: Boolean! = true, content: String) on 
+    ARGUMENT_DEFINITION | ENUM | ENUM_VALUE | FIELD_DEFINITION | INPUT_FIELD_DEFINITION 
+    | INPUT_OBJECT | INTERFACE | OBJECT | SCALAR | SCHEMA | UNION
+```
+
+The directive can be defined on any type system directive location (all of which allow a description to be defined).
+
+## Arguments
+
+The directive definition proposes two arguments.
+
+### public
+- Required but defaults to `true`.
+- Declares whether the description is to be propagated in the federated graph.
+- On any given instance of a shared definition, only one `@openfed__setDescription` instance may set `public` to
+`true` or there will be a composition error.
+- If `public` is set to `false`, the description will _never_ be propagated to the federated graph.
+This means the description instance will effectively be completely ignored by federation, 
+_e.g._, even when following "normal" WunderGraph Cosmo description rules.
+
+### content
```

Comment:
should it be named `overrideContent` or just `override`?

it will make more verbose what it stands for, without looking into directive definition

Reply:
Great suggestion!